packages/monorepo-scripts/src/publish.ts

```diff
@@ -107,30 +113,65 @@ package.ts. Please add an entry for it and try again.`,
     }
 }
 
+async function checkPublishRequiredSetupAsync(): Promise<boolean> {
+    // check to see if logged into npm before publishing
+    try {
+        await execAsync(`sudo npm whoami`);
```

Comment:
Why the `sudo` here?

Reply:
Without it, the `npm whoami` won't pick up the users credentials... I'm guessing running `npm login` runs in the sudo userspace? Not entirely sure though.

Reply:
Is that environment specific and dependent on the way npm was installed and logged in? We don't run the further publish tasks as `sudo` so it seems a bit strange. My `npm whoami` returns correctly.

Reply:
Not sure why it won't work without sudo. I checked and the script is running in the same user space as my command line. Will add a hack comment and punt on this for now.